src/lib/ndpi_main.c

```diff
@@ -3424,12 +3421,9 @@ ndpi_protocol ndpi_detection_process_packet(struct ndpi_detection_module_struct
   if((ret.protocol == NDPI_PROTOCOL_UNKNOWN)
      && flow->packet.iph
      && (!flow->host_already_guessed)) {
-    struct in_addr pin;
     
-    pin.s_addr = flow->packet.iph->saddr;
-    if((flow->guessed_host_proto_id = ndpi_network_ptree_match(ndpi_struct, &pin)) == NDPI_PROTOCOL_UNKNOWN) {
-      pin.s_addr = flow->packet.iph->daddr;
-      flow->guessed_host_proto_id = ndpi_network_ptree_match(ndpi_struct, &pin);
+    if((flow->guessed_host_proto_id = ndpi_network_ptree_match(ndpi_struct, (struct in_addr *)&flow->packet.iph->saddr)) == NDPI_PROTOCOL_UNKNOWN) {
```

Comment:
Since we are not writing to pin anymore we can pass the iph field in directly, no need to extract it first anymore.